### PR TITLE
Print Passed and Failed methods in detailed test summary

### LINKED ISSUE #18685: Print successful test methods when running `bazel test --test_summary=detailed`
### Description of the feature request:

Print the results of each test method when running the test module.

e.g. 
zipalign_tests (6 Tests)
[1/6] Align#Unaligned: PASSED (0ms)
[2/6] Align#DoubleAligment: PASSED (0ms)
[3/6] Align#Holes: PASSED (0ms)
[4/6] Align#DifferenteOrders: PASSED (0ms)
[5/6] Align#DirectoryEntryDoNotRequireAlignment: PASSED (0ms)
[6/6] Align#DirectoryEntry: PASSED (0ms)

### What underlying problem are you trying to solve with this feature?

Currently, regardless of any options (--test_summary, --test_output), Bazel's output does not print the passed test methods.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 6.2.1

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
92619b01fa7bbaf14ea1ceab3c48033e58d83eb3
```

### Have you found anything relevant by searching the web?

No

### Any other information, logs, or outputs that you want to share?

Reproduce steps:
In https://github.com/bazelbuild/examples/tree/main/third-party-dependencies
Run bazel test :gtest_test
```
//:gtest_test                                                            PASSED in 0.0s
```
Cant not show any passed test methods.
expects:
```
//:gtest_test                                                            PASSED in 0.0s  
     [PASSED] SumbNegativeValues  (0.0s)
     [PASSED] SumPositiveValues  (0.0s)
```

### PR DESCRIPTION
**Pull request by NelsonLi0701**
Previously, Bazel only printed Failed methods in the detailed test summary when using the `--test_summary=detailed` option. With this change, both Passed and Failed methods are now printed, providing a more comprehensive view of the test results.

Fix #18685

RELNOTES:  The `--test_summary=detailed` option now also prints passed test cases.

### CODE REVIEW

File: src/test/java/com/google/devtools/build/lib/runtime/TestSummaryTest.java [hunk 1]
```diff
@@ -518,6 +518,23 @@ public void testCollectingFailedDetails() throws Exception {
     verify(printer).print(find("ERROR.*cherry"));
   }
 
+  @Test
+  public void testCollectingPassedDetails() throws Exception {
```
Comment (jin) [inline src/test/java/com/google/devtools/build/lib/runtime/TestSummaryTest.java:522]: Can you add a test that has a mixture of FAILED, PASSED and ERROR cases and verify that all cases are printed? And update `testCollectingFailedDetails` to only check for failed/error cases.

**Review by jin — COMMENTED** [review]
Please also add an integration test like this: https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/bazel/bazel_test_test.sh;l=652-663;drc=ecf9b9a82aa231cd1fa9178dd9ab64dd67adaf25

**Comment by NelsonLi0701** [comment]
> Please also add an integration test like this: https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/bazel/bazel_test_test.sh;l=652-663;drc=ecf9b9a82aa231cd1fa9178dd9ab64dd67adaf25

Added `test_detailed_test_summary_for_passed_test` and rename the original `test_detailed_test_summary` to  `test_detailed_test_summary_for_failed_test`

**Comment by jin** [comment]
According to Bazel PR import instructions, I'll let the gTech team handle the PR import.

**Comment by jin** [comment]
Congrats on your first PR to Bazel Nelson!

**Pull request closed without merging**